src/main.ts

```diff
@@ -65,9 +65,24 @@ async function main() {
 
 	startUpPrompt(config.SUBSTRATE.URL, chainName.toString(), implName.toString());
 
+	const middlewares = [json(), middleware.httpLoggerCreate(logger)];
+
+	if (args.prometheus) {
+		// Create Metrics App
+		const metricsApp = new Metrics_App({
+			port: 9100,
+			host: config.EXPRESS.HOST,
+		});
+
+		// Generate metrics middleware
+		middlewares.push(metricsApp.middleware());
+		// Start the Metrics server
+		metricsApp.listen();
+	}
+
 	// Create our App
 	const app = new App({
-		preMiddleware: [json(), middleware.httpLoggerCreate(logger)],
+		preMiddleware: middlewares,
```

Comment:
Small nit -  but I think we can just call `middlewares` `preMiddleware`. That way its a bit more clear as to what value it holds.